### DISCUSSION TITLE
Function that gets executed when data is successfully queried from cache or first time

### DISCUSSION THREAD

**Discussion by pietroanello** [TanStack/query#7486] [Ideas]
I searched a lot for something like this, and I've read a lot about why the onSuccess callback was removed, but I haven't found any good alternative to execute code when data is retrieved from cache.

**Comment by TkDodo** [comment]
depends on what code you'd want to execute? Remember that reading from the cache is a synchronous action, so you would be _in render_ when that happens ...
